### PR TITLE
Step2

### PR DESCRIPTION
**Pull request by DZwell**

### CODE REVIEW

File: src/test_servers.py [hunk 1]
```diff
@@ -32,3 +44,38 @@ def test_response_error():
...
+
+def test_parse():
+    """Assert 404 error works correctly."""
+    from server import response_error
+    request = u'404 Page Not Found'
+    response = response_error(request)
+    response = response.split()
+    assert response[1] == '404'
+
+
```
Comment (tlake) [inline src/test_servers.py:81]: There doesn't seem to be a test checking the behavior of `parse_request()` when everything should be working perfectly.

File: src/server.py [hunk 2]
```diff
@@ -5,30 +5,51 @@
 import email.utils
 
 
-def response_ok():
+def parse_request(request):
+    """Parse and validate request to confirm parts are correct."""
+    lines = request.split('\r\n')
+    try:
+        lines[1][:3] == 'GET'
```
Comment (tlake) [inline src/server.py:12]: There are a couple of things wrong here.
1. Remember that things are zero-indexed in Python. If you're trying to access the first item in the list `lines`, you'll need to be using `lines[0]`.
2. This try/except block isn't fulfilling the purpose I think you were intending. Evaluating `lines[1][:3] == 'GET'` will result in one of three return values: `True`, `False`, or an exception. Note that the `except` block will only be entered if an exception is encountered. Let's say, for example, that `lines[1]` is a string, and the first three characters are `'WTF'`. Evaluating `lines[1][:3] == 'GET'` will return `False` (because 'WTF' != 'GET'), but `False` is not an error. Therefore, even though the HTTP request is not a GET request like we want it to be, the result of `False` won't take us to the except block, and the code intended to be run when the request method isn't 'GET' will not be.

tl;dr You should be refactoring these try/except blocks into if/elif/else blocks because you're currently only checking to see whether some sort of error is raised instead of checking to see whether a particular condition is met.

File: src/test_servers.py (hunk 1 above)
Comment (nadiabahrami) [inline src/test_servers.py:81]: Adressed
